src/daedalus.cpp

```diff
@@ -0,0 +1,291 @@
+// Copyright 2025 'daedalus' authors. See repository licence in LICENSE.md.
+
+// clang-format off
+#include <Rcpp.h>
+#include <RcppEigen.h>
+
+#include <algorithm>
+#include <cmath>
+#include <vector>
+#include <boost/numeric/odeint.hpp>
+// clang-format on
+
+/// @brief Model constants.
+const int N_AGE_GROUPS = 4L;
+const int N_ECON_SECTORS = 45L;
+const int N_GROUPS = N_AGE_GROUPS + N_ECON_SECTORS;
+const int iS = 0, iE = 1, iIs = 2, iIa = 3, iH = 4, iR = 5, iD = 6, iV = 7,
+          idE = 8, idH = 9;
+const int N_COMPARTMENTS = 10;
+
+const double excess_mortality_coef = 1.6;
+
+/// @brief State type for ODEs.
+typedef Eigen::MatrixXd state_type;
+
+/// @brief
+/// @tparam T
+/// @param value The value of the opennness coefficient. For an 80% closure of a
+/// sector, the openness coefficient would be 0.2.
+/// @param flag
+/// @return Either `value` when `flag` = 1.0, or 1.0 when `flag` = 0.0.
+template <typename T>
+T apply_npi(T value, const double &flag) {
```

Comment:
why not

```suggestion
T apply_npi(T value, const double flag) {
```

no need for a reference here if you're not mutating it (unlike below with things like Eigen::ArrayXd where passing a reference makes sense). See, for example https://stackoverflow.com/questions/7561690/why-do-we-not-pass-pod-by-reference-in-functions

Reply:
Sure - wasn't thinking too much about this.